### PR TITLE
add DeepONet with paddle backend

### PR DESCRIPTION
**Pull request by HydrogenSulfate**
add DeepONet with paddle backend

### CODE REVIEW

File: deepxde/backend/paddle/tensor.py [hunk 1]
```diff
@@ -9,6 +9,8 @@
 
 if paddle.device.is_compiled_with_cuda():
     paddle.device.set_device("gpu")
+else:
+    paddle.device.set_device("cpu")
```
Comment (lululxvi) [inline deepxde/backend/paddle/tensor.py:13]: Do we need this line? If no GPU available, would paddle automatically choose CPU?

File: deepxde/nn/paddle/__init__.py [hunk 2]
```diff
@@ -1,6 +1,7 @@
 """Package for paddle NN modules."""
 
-__all__ = ["DeepONetCartesianProd", "FNN"]
+__all__ = ["DeepONet", "DeepONetCartesianProd", "FNN"]
 
 from .deeponet import DeepONetCartesianProd
+from .deeponet import DeepONet
```
Comment (lululxvi) [inline deepxde/nn/paddle/__init__.py:6]:
```
from .deeponet import DeepONet
from .deeponet import DeepONetCartesianProd
```

File: deepxde/nn/paddle/deeponet.py [hunk 3]
```diff
@@ -3,6 +3,102 @@
 from .fnn import FNN
 from .nn import NN
 from .. import activations
+from ..import initializers
```
Comment (lululxvi) [inline deepxde/nn/paddle/deeponet.py:6]:
```python
from .. import initializers
```

File: deepxde/nn/paddle/deeponet.py [hunk 4]
```diff
@@ -3,6 +3,102 @@
...
+        self.layer_sizes_loc = layer_sizes_trunk
+
+        if isinstance(activation, dict):
+            self.activation_branch = activations.get(activation["branch"])
+            self.activation_trunk = activations.get(activation["trunk"])
+        else:
+            activation_branch = self.activation_trunk = activations.get(activation)
+
+        self.kernel_initializer = initializers.get(kernel_initializer)
+        self.stacked = stacked
```
Comment (lululxvi) [inline deepxde/nn/paddle/deeponet.py:54]: `self.stacked` is not used in the following code. Delete `self.stacked` and related codes.

File: deepxde/nn/paddle/deeponet.py [hunk 5]
```diff
@@ -3,6 +3,96 @@
...
+            self.branch = FNN(layer_sizes_branch, activation_branch, kernel_initializer)
+        self.trunk = FNN(layer_sizes_trunk, self.activation_trunk, kernel_initializer)
+        self.use_bias = use_bias
+        if use_bias:
+            # register bias to parameter for updating in optimizer and storage
+            self.b = self.create_parameter(
+                shape=(1, ),
+                default_initializer=initializers.get("zeros")
+            )
+        self.trainable_branch = trainable_branch
```
Comment (lululxvi) [inline deepxde/nn/paddle/deeponet.py:68]: The current code doesn't use `trainable_branch` and `trainable_trunk`. Delete them.

Comment (HydrogenSulfate) [reply deepxde/nn/paddle/deeponet.py]: deleted

File: deepxde/nn/paddle/deeponet.py [hunk 6]
```diff
@@ -3,6 +3,90 @@
...
+            self.branch = FNN(layer_sizes_branch, activation_branch, kernel_initializer)
+        self.trunk = FNN(layer_sizes_trunk, self.activation_trunk, kernel_initializer)
+        self.use_bias = use_bias
+        if use_bias:
+            # register bias to parameter for updating in optimizer and storage
+            self.b = self.create_parameter(
+                shape=(1, ),
+                default_initializer=initializers.get("zeros")
+            )
+        self.regularizer = regularization
```
Comment (lululxvi) [inline deepxde/nn/paddle/deeponet.py:64]: `self.regularizer` never used. Delete.

File: deepxde/nn/paddle/deeponet.py [hunk 7]
```diff
@@ -3,6 +3,90 @@
...
+                "Output sizes of branch net and trunk net do not match."
+            )
+        x = paddle.einsum("bi,bi->b", x_func, x_loc)
+        # Add bias
+        if self.use_bias:
+            x += self.b
+        if self._output_transform is not None:
+            x = self._output_transform(inputs, x)
+        # Avoid incorrect shape broadcast when calculating loss
+        if x.ndim == 1:
```
Comment (lululxvi) [inline deepxde/nn/paddle/deeponet.py:87]: `x` has shape batch-size x N, and `x.ndim == 1:` is never `True`.

File: deepxde/nn/paddle/deeponet.py (hunk 6 above)
Comment (HydrogenSulfate) [reply deepxde/nn/paddle/deeponet.py]: sorry I close this pull request by mistake, please refer to the new pull request https://github.com/lululxvi/deepxde/pull/935

File: deepxde/nn/paddle/deeponet.py (hunk 7 above)
Comment (HydrogenSulfate) [reply deepxde/nn/paddle/deeponet.py]: the result after einsum is [batch_size, ] but not [batch_size, N], so `reshape` is necessary

**Pull request closed without merging**